appinventor/components/src/com/google/appinventor/components/runtime/PhoneCall.java

```diff
@@ -277,42 +308,34 @@ public void onReceive(Context context, Intent intent) {
       if(TelephonyManager.ACTION_PHONE_STATE_CHANGED.equals(action)){
         String state = intent.getStringExtra(TelephonyManager.EXTRA_STATE);
         if(TelephonyManager.EXTRA_STATE_RINGING.equals(state)){
-          // Incoming call rings
-          status = 1;
+          status = CallStatus.INCOMING_WAITING;
           number = intent.getStringExtra(TelephonyManager.EXTRA_INCOMING_NUMBER);
           if (number == null) {
             // This gets called first with null, then with the actual number.
             // Ignore the first invocation.
             return;
           }
-          PhoneCallStarted(1, number);
-        }else if(TelephonyManager.EXTRA_STATE_OFFHOOK.equals(state)){
-          // Call off-hook
-          if(status == 1){
-            // Incoming call answered
-            status = 3;
+          PhoneCallStartedAbstract(StartedStatus.Incoming, number);
+        } else if (TelephonyManager.EXTRA_STATE_OFFHOOK.equals(state)){
+          if(status == CallStatus.INCOMING_WAITING){
```

Comment:
Code style: space between `if` and `(`. Note you can run `ant checkstyle` before committing changes to see if any rules are violated.